analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ConditionEvaluatesToConstant.CSharp7.cs

```diff
@@ -252,17 +252,24 @@ public int DoSomething(object value)
             if (!(value is bool valueBool))
```

Comment:
could you please also add a test with 
```
    if (x is (string s, int i))
    {
        if (s == null)  // Noncompliant
        {
        }
    }
```
inside `SonarAnalyzer.UnitTest\TestCases\ConditionEvaluatesToConstant.CSharp8.cs`

Reply:
and also
```
        if (x is (((string s1, (string  s2, String s3)), string s), int i))
        {
            if (s3 == null) // Noncompliant
            {
            }
        }
```

And maybe a test with generated code with inner tuples to see if it throws at some point? Or maybe that's a good test for the nested expression stress test?

Reply:
It's little more complicated. `x is(string s, int i)` is C# 7 and it's not `ParanthesizedVariableDesignation`. It's recursive pattern.

I covered this in first review commit. I'll add the recursive hell :)

Reply:
thanks!